### ISSUE TITLE
Error categorising an XHTML page as a TextPage rather than HtmlPage

### ISSUE THREAD

**Issue by antosarho** [HtmlUnit/htmlunit#663]
DefaultPageCreator.determinePageType will often categorise the mime type of a page coded as XHTML as Text rather than HTML.

Here's my example page:
```
<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="lt" lang="lt">
<head>
    <script>
        var article = {
            "summary": "<p>​Koronaviruso atvejų &scaron;uoliai priverčia žmones sunerimti. Vis dažniau dairomasi, kur greičiau atlikti koronavirusui nustatyti skirtą testą. Tiesa, jų esama kelių skirtingų rū&scaron;ių. Todėl pacientams kyla daugybė klausimų.</p>\n"
          }
   </script>
</head>
<body>
<h1>Some Stuff</h1>
</body>
</html>
```

The above page will get categorised as Text rather than HTML, which is a mistake to me.

In version 2.46, that was categorised as HTML.

From version 2.47 onwards, when different HTML sniffing code was implemented, that gets categorised as Text.

**Comment by BrillBay** [comment]
@antosarho the determination is done based on content type - do you know the content type sent by the server

**Comment by rbri** [comment]
@antosarho i think i found another possible reason (in this case a bug) for that - can you please check if the response content starts with a bom?

**Comment by rbri** [comment]
@antosarho did some tests but if there is a bom header the browsers are handling this as text page
the reason must be somewhere else

**Comment by antosarho** [comment]
I see the problem. So the server isn't specifying the content type.

Because the content type isn't being specified, `DefaultPageCreator.determinePageType(WebResponse webResponse)` tries to determine the content type itself. 

And in there, from version 2.7 onwards, the derivation of the content type fails for the code specified above.

The `htmlPatterns` are fine, it's just that the `asAsciiString` has to start with one of the html patterns, which the code above doesn't do because it starts with the XML specification.

In version 2.6, it was much simpler and just checked this: 
```
            if (asAsciiString.contains("<HTML")) {
                return MimeType.TEXT_HTML;
            }
```

Cheers

**Comment by rbri** [comment]
@antosarho yep found the same thing, working on that later today. But i think the result should be an xhtml page in this case

**Comment by antosarho** [comment]
Yep, XHTML page is even more accurate.

Thanks

**Comment by rbri** [comment]
@antosarho Hope you like the fix - made a new snapshot build 3.8.0-SNAPSHOT available.

Please test.

**Comment by antosarho** [comment]
@rbri thanks muchly -- works well!
